jenkins/ci.suse.de/pipelines/openstack-ardana-gerrit.Jenkinsfile

```diff
@@ -58,15 +93,55 @@ pipeline {
               string(name: 'develproject', value: "$develproject"),
               string(name: 'repository', value: "$repository")
           ], propagate: false, wait: true
-          def jobResult = slaveJob.getResult()
-          def jobUrl = slaveJob.buildVariables.blue_ocean_buildurl
+          env.jobResult = slaveJob.getResult()
+          env.jobUrl = slaveJob.buildVariables.blue_ocean_buildurl
           def jobMsg = "Build ${jobUrl} completed with: ${jobResult}"
           echo jobMsg
-          if (jobResult != 'SUCCESS') {
+          sh '''
+            echo "- ${STAGE_NAME}: ${jobResult} (${jobUrl})" >> results.txt
+          '''
+          if (env.jobResult != 'SUCCESS') {
              error(jobMsg)
```

Comment:
Should we have something like this for the commit message validation to avoid running this stage on failure?

Reply:
Let's see if I correctly understand your concern: you would like to avoid running the `integration test` stage if a failure occurs during the `validate commit message` stage ? If that's the case, then the pipeline's default behavior will do that for use, we don't need to implement anything special to enable that.

Reply:
ah, right you are. I got confused then as I am unsure what the `error()` method is doing here?

Reply:
I'm using `error()` here to propagate a failure in the downstream job run. I've tried several solutions to this problem, like using `try/catch` or stage `post` blocks, but the only way that allows me to print the URL pointing to the failed downstream build is to use `propagate: false` when triggering the downstream build and then analyze the result (see #2857 and #2843 for more info).

Reply:
Ah okay, I think I follow now.

As a side note, why is the "validate commit message" stage fully defined in this JenkinsFile rather than being its own slaveJob so that it is runnable independently and all "jobs" for a pipeline are called the same way?